Source/misdat.cpp

```diff
@@ -127,68 +129,162 @@ MissileDataStruct MissileData[] = {
-	{ "ex_red3",  MFILE_EXRED3,      1, 1, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, {  7,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 }, { 292,   0,   0,   0,   0,   0,   0,   0,  0,  0,  0,  0,  0,  0,  0,  0 }, { 114,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 } },
-	{ "ms_blb",   MFILE_BONEDEMON,  16, 1, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, { 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15 }, {  96,  96,  96,  96,  96,  96,  96,  96, 96, 96, 96, 96, 96, 96, 96, 96 }, {   8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8 } },
-	{ "ex_ora1",  MFILE_EXORA1,      1, 1, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, { 13,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 }, {  96,   0,   0,   0,   0,   0,   0,   0,  0,  0,  0,  0,  0,  0,  0,  0 }, { -12,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 } },
-	{ "ex_blu3",  MFILE_EXBL3,       1, 1, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, {  7,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 }, { 292,   0,   0,   0,   0,   0,   0,   0,  0,  0,  0,  0,  0,  0,  0,  0 }, { 114,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 } },
-	{ "",         MFILE_NONE,        0, 0, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, {  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 }, {   0,   0,   0,   0,   0,   0,   0,   0,  0,  0,  0,  0,  0,  0,  0,  0 }, {   0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 } },
+	// name,      animName,        animFAmt, flags,                animDelay[16], animLen[16],                           animWidth[16],          animWidth2[16]
+	{ "Arrows",   MFILE_ARROWS,     1,       MFLAG_LOCK_ANIMATION,    0,            16,                                     96,                     16               },
+	{ "Fireba",   MFILE_FIREBA,    16,       0,                       0,            14,                                     96,                     16               },
+	{ "Guard",    MFILE_GUARD,      3,       0,                       1,          { 15, 14,  3 },                           96,                     16               },

```

Comment:
It looks like it would also be possible to split up the unusual parts of `Guard`, `Firewal`, `Portal`, `Holy`, `Acidpud`, and `Sklball` if we modefy `missiles_process_charge()` to be aware of the mapping in cases where `mis->_mimfnum > animFAmt`

```cpp
MissileExtraFileData MissileExtraSpriteData[] = {
{ "Guard2", 3, 0, 1, 14, 96, 16 },
{ "Guard3", 3, 0, 1, 14, 96, 16 },
{ "Firewal2", 2, 0, 0, 11, 128, 32 },
{ "Portal2", 2, 0, 1, 16, 96, 16 },
{ "Holy2", 16, 0, 0, 14, 96, 16 },
{ "Acidpud2", 2, 0, 0, 4, 96, 16 },
{ "Sklball9", 9, 0, 1, 8, 96, 16 },
};
```

Additionally this also would need a special version of `SetMissAnim()` to be used in the place where the 7 special missile animations are loaded using `SetMissDir()`.

But I think this is optional in relation to this PR.